packages/backend/src/graphql/mutations/login-with-sgid.ts

```diff
@@ -0,0 +1,108 @@
+import { SgidClient } from '@opengovsg/sgid-client'
+
+import appConfig from '@/config/app'
+import { getOrCreateUser, setAuthCookie } from '@/helpers/auth'
+import { validateAndParseEmail } from '@/helpers/email-validator'
+import type Context from '@/types/express/context'
+
+const sgidClient = new SgidClient({
+  clientId: appConfig.sgid.clientId,
+  clientSecret: appConfig.sgid.clientSecret,
+  privateKey: appConfig.sgid.privateKey,
+  redirectUri: `${appConfig.webAppUrl}/login/sgid/redirect`,
+})
+
+interface PublicOfficerEmployment {
+  workEmail: string | null
+  agencyName: string | null
+  departmentName: string | null
+  employmentType: string | null
+  employmentTitle: string | null
+}
+
+interface LoginWithSgidParams {
+  input: { authCode: string; nonce: string; verifier: string }
+}
+
+interface LoginWithSgidResult {
+  nextUrl: string
+}
+
+async function parsePocdexEmployments(
+  rawData: string,
+): Promise<PublicOfficerEmployment[]> {
+  const allEmployments = (JSON.parse(rawData) as PublicOfficerEmployment[])
```

Comment:
u might wanna try catch this and provide a more useful error message

Reply:
referring to the JSON.parse

Reply:
done, I thinking to switch to zod in a later PR.